pkg/metrics/agent.go

```diff
@@ -81,6 +81,10 @@ func (c *Config) ApplyDefaults() error {
 		return errors.New("cannot use configs when scraping_service mode is enabled")
 	}
 
+	if c.DisableKeepAlives && c.IdleConnTimeout > 0 {
+		return errors.New("cannot disable keep-alives and set the idle connection timeout at the same time")
+	}
```

Comment:
I think this might cause more confusion than is necessary since if you only set `http_disable_keepalives: true` it'll fail because of the default idle conn timeout. 

We can just document that `http_idle_conn_timeout` is ignored when `http_disable_keepalives` is enabled IMO.